File: rajawali/src/main/java/org/rajawali3d/renderer/AFrameTask.java
package org.rajawali3d.renderer;

import timber.log.Timber;

/**
 * @author Jared Woolston (jwoolston@tenkiv.com)
 */
public abstract class AFrameTask implements Runnable {

    protected abstract void doTask();

    @Override
    public void run() {
        try {
            doTask();
        } catch (Exception e) {
            Timber.e("Execution Failed: " + e.getMessage());
            e.printStackTrace();
        }
    }
}

File: rajawali/src/main/java/org/rajawali3d/renderer/AFrameTask.kt
package org.rajawali3d.renderer

import timber.log.Timber

abstract class AFrameTask : Runnable {
    protected abstract fun doTask()

    override fun run() {
        try {
            doTask()
        } catch (e: Exception) {
            Timber.e("Execution Failed: ${e.message}")
        }
    }
}
